libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/agent_application.py

```diff
@@ -233,18 +275,17 @@ def __call(func: RouteHandler[StateT]) -> RouteHandler[StateT]:
             logger.debug(
                 f"Registering activity handler for route handler {func.__name__} with type: {activity_type} with auth handlers: {auth_handlers}"
             )
-            self._routes.append(
-                Route[StateT](__selector, func, auth_handlers=auth_handlers)
-            )
+            self.add_route(__selector, func, auth_handlers=auth_handlers, **kwargs)
             return func
 
         return __call
 
     def message(
```

Comment:
These overloaded names are confusing. In [C# this is "OnActivity"](https://github.com/microsoft/Agents-for-net/blob/a734eb543c2165fb28b421b70c8335ef4394b807/src/libraries/Builder/Microsoft.Agents.Builder/App/AgentApplication.cs#L178) which is a less confusing name. 

EDIT: Sorry - this one is "onMessage". They should all mirror the C# code here if possible, as this naming seem is (at best) hard to follow.